client/my-sites/plugins/jetpack-plugins-setup/index.jsx

```diff
@@ -299,6 +299,11 @@ const PlansSetup = React.createClass( {
 					{ translate( 'Manual Installation' ) }
 				</NoticeAction>
 			);
+			if ( plugin.error.code === 'already_registered' ) {
+				statusProps.status = 'is-info';
+				statusProps.text = translate( 'This plugin is already registered with another plan.' );
+				delete statusProps.children;
+			}
```

Comment:
We can leave this for another PR but this logic is a bit difficult for me to follow.

What do folks think of breaking this into two functions `getPluginErrorStatus` and `getPluginStatus` where we return jsx within each switch case instead of falling through?